### PR TITLE
TST/CLN/BUG: Fix corr nearest factor

### PR DESCRIPTION
**Pull request by bashtage**
Fix test that wasn't testing anything
Pass tuning parameters into optimization function
Increase max iterations

xref #5197

### CODE REVIEW

File: statsmodels/stats/tests/test_corrpsd.py [hunk 1]
```diff
@@ -315,42 +315,37 @@ def test_corr_nearest_factor(self):
...
-            assert_equal(rslt.Converged, True)
-            mat_sparse = rslt.corr.to_matrix()
+        rslt = corr_nearest_factor(mat, dm, maxiter=10000)
+        assert_equal(rslt.Converged, True)
+        mat_dense = rslt.corr.to_matrix()
 
-            assert_allclose(mat_dense, mat_sparse, rtol=0.25,
-                            atol=1e-3)
+        rslt = corr_nearest_factor(smat, dm, maxiter=10000)
+        assert_equal(rslt.Converged, True)
```
Comment (jbrockmendel) [inline statsmodels/stats/tests/test_corrpsd.py]: If we can get away with a plain assertion that’d be nice.

**Comment by jbrockmendel** [comment]
+1

File: statsmodels/stats/correlation_tools.py [hunk 2]
```diff
@@ -205,14 +204,13 @@ def cov_nearest(cov, method='clipped', threshold=1e-15, n_fact=100,
...
-
-    '''
+    """
 
     from statsmodels.stats.moment_helpers import cov2corr, corr2cov
     cov_, std_ = cov2corr(cov, return_std=True)
     if method == 'clipped':
         corr_ = corr_clipped(cov_, threshold=threshold)
-    elif method == 'nearest':
+    else:  # method == 'nearest'
```
Comment (josef-pkt) [inline statsmodels/stats/correlation_tools.py]: better keep `elif` and raise ValueError on an `else`

**Comment by josef-pkt** [comment]
AFAICS, the important change is BUG fix (missing keyword propagation)
which test didn't test anything?

In general: keeping equivalent refactoring (parameterized unit tests) in separate commit fromm bug or test fixes makes a PR easier to review and understand.

LGTM but I didn't see whether the changes in the unit tests are just dedenting or substantive

File: statsmodels/stats/tests/test_corrpsd.py [hunk 3]
```diff
@@ -283,74 +276,67 @@ def test_corr_nearest_factor_arrpack(self):
...
-            # Threshold it
-            mat *= (np.abs(mat) >= 0.35)
-            smat = sparse.csr_matrix(mat)
+        # Get the correlation matrix
+        _project_correlation_factors(X)
+        X *= 0.7
+        mat = np.dot(X, X.T)
+        np.fill_diagonal(mat, 1)
 
-            rslt = corr_nearest_factor(smat, dm)
```
Comment (bashtage) [inline statsmodels/stats/tests/test_corrpsd.py]: `smat` here

File: statsmodels/stats/tests/test_corrpsd.py [hunk 4]
```diff
@@ -283,74 +276,67 @@ def test_corr_nearest_factor_arrpack(self):
...
-            assert_equal(rslt.Converged, True)
-            mat_dense = rslt.corr.to_matrix()
+        # Threshold it
+        mat *= (np.abs(mat) >= 0.35)
+        smat = sparse.csr_matrix(mat)
 
-            rslt = corr_nearest_factor(smat, dm)
```
Comment (bashtage) [inline statsmodels/stats/tests/test_corrpsd.py]: also `smat` on needs to be mat to compare dense to sparse

**Comment by bashtage** [comment]
The only unit test change is replacing one `smat` with a dense array.  Everything else is moving to parametrize so that failures can be isolated on the dimension by removing the for loop.

**Comment by josef-pkt** [comment]
all green, merging

@bashtage Thank you

**Pull request merged**